include/revng/Pipeline/Step.h

```diff
@@ -27,6 +27,11 @@
 
 namespace pipeline {
 
+struct StepArtifactsInfo {
```

Comment:
this struct is only used inside a Step and step does not return a direct reference or pointer to it from in any method,
so i would push it inside the class step as private struct declaration

```
class Step {
...
private:
struct ArtifactInfo {
...
};
...
};

```